src/bin/cargo-plumbing/plumbing/read_manifest.rs

```diff
@@ -0,0 +1,26 @@
+use std::path::PathBuf;
+
+use cargo::{CargoResult, GlobalContext};
+use cargo::core::{EitherManifest, SourceId};
+use cargo::util::toml::read_manifest;
+
+#[derive(Debug, clap::Args)]
+pub(crate) struct Args {
+    manifest_path: PathBuf,
+}
+
+pub(crate) fn exec(gctx: &GlobalContext, args: Args) -> CargoResult<()> {
+    let manifest_path = args.manifest_path.canonicalize()?;
```

Comment:
`canonicalize` is banned within the cargo code base.  They use `GlobalContext` to get the current dir to fully qualify a path